services/src/main/java/org/keycloak/services/resources/account/AccountConsole.java

```diff
@@ -92,10 +96,31 @@ public Object getResource() {
     public void close() {
     }
 
+    protected Response redirectToLogin(String path) {
+
+        var oauthRedirect = new AbstractSecuredLocalService.OAuthRedirect();
+        oauthRedirect.setAuthUrl(OIDCLoginProtocolService.authUrl(session.getContext().getUri()).build(realm.getName()).toString());
+        oauthRedirect.setClientId(client.getClientId());
+        oauthRedirect.setSecure(realm.getSslRequired().isRequired(session.getContext().getConnection()));
+
+        UriBuilder accountUriBuilder = Urls.accountBase(session.getContext().getUri().getBaseUri());
+        if (!StringUtil.isNullOrEmpty(path)) {
+            accountUriBuilder.path(path);
+        }
+        URI accountUri = accountUriBuilder.build(realm.getName());
+
+        return oauthRedirect.redirect(session.getContext().getUri(), accountUri.toString());
+    }
+
     @GET
     @NoCache
-    @Path("{any:.*}")
-    public Response getMainPage() throws IOException, FreeMarkerException {
+    @Path("{path:.*}")
+    public Response getMainPage(@PathParam("path") String path) throws IOException, FreeMarkerException {
+
+        if (auth == null) {
+            return redirectToLogin(path);
+        }
```

Comment:
Could we add a test for this redirect logic?

Reply:
I just added a test for this.